cpp/server/ct-mirror.cc

```diff
@@ -331,40 +295,19 @@ int main(int argc, char* argv[]) {
 
   Server::StaticInit();
 
-  if (!FLAGS_sqlite_db.empty() + !FLAGS_leveldb_db.empty() +
-          (!FLAGS_cert_dir.empty() | !FLAGS_tree_dir.empty()) !=
-      1) {
-    std::cerr << "Must only specify one database type.";
-    exit(1);
-  }
-
-  if (FLAGS_sqlite_db.empty() && FLAGS_leveldb_db.empty()) {
-    CHECK_NE(FLAGS_cert_dir, FLAGS_tree_dir)
-        << "Certificate directory and tree directory must differ";
-  }
-
-  Database* db;
-
-  if (!FLAGS_sqlite_db.empty()) {
-    db = new SQLiteDB(FLAGS_sqlite_db);
-  } else if (!FLAGS_leveldb_db.empty()) {
-    db = new LevelDB(FLAGS_leveldb_db);
-  } else {
-    db = new FileDB(new FileStorage(FLAGS_cert_dir, FLAGS_cert_storage_depth),
-                    new FileStorage(FLAGS_tree_dir, FLAGS_tree_storage_depth),
-                    new FileStorage(FLAGS_meta_dir, 0));
-  }
+  ServerHelper::EnsureValidatorsRegistered();
+  unique_ptr<Database> db(ServerHelper::ProvideDatabase());
+  CHECK(db != nullptr) << "No database instance created, check flag settings";
 
   const bool stand_alone_mode(FLAGS_etcd_servers.empty());
   const shared_ptr<libevent::Base> event_base(make_shared<libevent::Base>());
   ThreadPool internal_pool(8);
   UrlFetcher url_fetcher(event_base.get(), &internal_pool);
 
-  const std::unique_ptr<EtcdClient> etcd_client(
-      stand_alone_mode
-          ? new FakeEtcdClient(event_base.get())
-          : new EtcdClient(&internal_pool, &url_fetcher,
-                           SplitHosts(FLAGS_etcd_servers)));
+  const unique_ptr<EtcdClient> etcd_client(
+      stand_alone_mode ? new FakeEtcdClient(event_base.get())
+                       : new EtcdClient(&internal_pool, &url_fetcher,
+                                        SplitHosts(FLAGS_etcd_servers)));
```

Comment:
I approve of removing that `std::`, but I disapprove of not doing it separately. :wink: